File: frontend/src/types/telegram.ts
export type TelegramDestinatario = {
  id_destinatario: number;
  nombre: string;
  chat_id: string;
  activo: boolean;
  recibe_documentos: boolean;
  recibe_mantenimiento: boolean;
  recibe_viajes: boolean;
  created_at: string;
  updated_at: string;
};

export type CreateTelegramDestinatarioPayload = {
  nombre: string;
  chat_id: string;
  activo: boolean;
  recibe_documentos: boolean;
  recibe_mantenimiento: boolean;
  recibe_viajes: boolean;
};

export type UpdateTelegramDestinatarioPayload = Partial<CreateTelegramDestinatarioPayload>;

export type TelegramTestResponse = {
  enviado: boolean;
  mensaje: string;
  destinatario: TelegramDestinatario;
};

export type ProcesarNotificacionesResponse = {
  alertas_evaluadas: number;
  alertas_enviadas: number;
  alertas_sin_destinatario: number;
  alertas_omitidas: number;
  alertas_fallidas: number;
};

File: backend/app/schemas/telegram.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class TelegramDestinatarioBase(BaseModel):
    nombre: str
    chat_id: str
    activo: bool = True
    recibe_documentos: bool = True
    recibe_mantenimiento: bool = True
    recibe_viajes: bool = True


class TelegramDestinatarioCreate(TelegramDestinatarioBase):
    pass


class TelegramDestinatarioUpdate(BaseModel):
    nombre: str | None = None
    chat_id: str | None = None
    activo: bool | None = None
    recibe_documentos: bool | None = None
    recibe_mantenimiento: bool | None = None
    recibe_viajes: bool | None = None


class TelegramDestinatarioResponse(TelegramDestinatarioBase):
    id_destinatario: int
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class TelegramTestResponse(BaseModel):
    enviado: bool
    mensaje: str
    destinatario: TelegramDestinatarioResponse


class ProcesarNotificacionesResponse(BaseModel):
    alertas_evaluadas: int
    alertas_enviadas: int
    alertas_sin_destinatario: int
    alertas_omitidas: int
    alertas_fallidas: int
